### ISSUE TITLE
segmentation models - make `num_classes` and `softmax` non must

### ISSUE THREAD

**Issue by yonigottesman** [keras-team/keras-cv#2476]
Currently all the segmentation models `deep_lab_v3`, `segformer` etc.. create the final layer with `num_classes` and always does `softmax`. Maybe we should have an `activation` parameter and if its `None` dont do any activation. 
Some classification models require `sigmoid` and sometimes I want the unet for generative tasks without and activation

**Comment by sachinprasadhs** [comment]
Thanks for reporting the issue.
We have the re enhanced DeepLab and other models in Keras Hub which has a option to provide the `activation` .
For more details check https://github.com/keras-team/keras-hub/blob/master/keras_hub/src/models/deeplab_v3/deeplab_v3_segmenter.py
